ykcapi/src/lib.rs

```diff
@@ -48,6 +52,122 @@ pub extern "C" fn __ykutil_get_llvmbc_section(res_addr: *mut *const c_void, res_
     }
 }
 
+/// Reads out registers spilled to the stack of the previous frame during the deoptimisation
+/// routine. The order of the registers are in accordance to the DWARF register number mapping
+/// referenced in the SystemV ABI manual (https://uclibc.org/docs/psABI-x86_64.pdf).
+struct Registers {
+    addr: *const usize,
+}
+
+impl Registers {
+    /// Creates a Registers struct from a given a pointer on the stack containing spilled
+    /// registers.
+    fn from_ptr(ptr: *const c_void) -> Registers {
+        Registers {
+            addr: ptr as *const usize,
+        }
+    }
+
+    /// Read the spilled register value at the offset `off` from the previous stack frame.
+    fn read_from_stack(&self, off: isize) -> usize {
```

Comment:
This should be unsafe. This can be used to read arbitrary (including uninitialized and unmapped) memory.

Reply:
I assume you mean that I should mark the function as unsafe rather than wrapping the body in an unsafe block?

Reply:
Yes

Reply:
Thanks! Fixed in 04406d1